### ISSUE TITLE
5.0-pre8 heavy performance degradation

### ISSUE THREAD

**Issue by myblindy** [opentk/opentk#1534]
### Description

After upgrading a project to use `5.0-pre8` from `5.0-pre7`, with no other changes, performance dropped by more than two thirds. On this particular machine, it went from over 60 fps to ~30fps.

### Repro steps

You can try it yourself by cloning https://github.com/myblindy/tweey and checking out commit `c981418d4b00a74c72402c61e61c5c850dfeaf7b`. That one is using `5.0-pre7`. Changing it to `5.0-pre8` in all projects will tank the frame rate.

You will need the latest VS and .Net 7 installation to compile this, since it makes heavy use of source generators. 

### Related information 

* Operating system: Windows 10
* SDL or Native backend: OpenTK 5.0, so N/A
* What version of OpenTK: Nuget, `5.0-pre8`
* What runtime: .Net 7
* Workarounds: Wish I had any.

**Comment by NogginBops** [comment]
Have you tried profiling between the two changes?

**Comment by utkumaden** [comment]
I just remembered a very handy tool is built into git called `bisect`, and the notification made me come back to this thread. It can help you find the exact commit that introduced the regression.

Start by checking out 5.0-pre8, then using the following bisect commands.
```bash
git checkout 5.0-pre8
git bisect start
git bisect bad
git bisect good 5.0-pre7
```

After this, git will checkout a version for you to try. You can compile it and add OpenTK to a local nuget repository and try it with your application. If the regression persists, use `git bisect bad` or `git bisect good` in order to move to the next commit in the search.

You should be able to find the exact commit that introduced the issue and it might shine light into what the issue actually is.

**Comment by myblindy** [comment]
@NogginBops I did, yeah. It looks like the `Run` method simply isn't being called as often, probably some kind of timing issue? Looking through the latest commits around that method I saw that a previous commit introduced some sleep mechanism between calls, I'd be willing to bet that's the problem.

## pre-7:

![image](https://user-images.githubusercontent.com/541998/202758096-62aa0014-6981-4b59-a894-2039d33bedcf.png)

## pre-8:

![image](https://user-images.githubusercontent.com/541998/202758196-de86efd2-01b2-4313-ae7f-6bb714356860.png)

All code is identical between tries.

@utkumaden that only helps if all commits are equal, ie if you squash merged branches into single commits. If instead you merge individual commits, what you end up with is that half of them don't even compile, and half of the rest have big bugs or unimplemented branches (like `if`-branches, not git-branches).

**Comment by NogginBops** [comment]
@myblindy can you show how you create your window? Do you have vsync enabled?

**Comment by myblindy** [comment]
@NogginBops You can see it in the repository I linked, here's the initialization code: https://github.com/myblindy/tweey/blob/master/tweey/Program.cs#L8

To answer your second question, no, VSync is disabled. 

Note that #1537 is showing the exact same problem I have but for the 4.x branch. The rendering loop is definitely borked in these versions.

**Comment by NogginBops** [comment]
Closing this as it should be fixed with #1659 as that merges the latest `4.x` (which contains #1571) into `5.0.0-pre9`.
